pallets/subscriptions/src/types.rs

```diff
@@ -0,0 +1,30 @@
+use frame_support::pallet_prelude::*;
+
+/// Subscription settings for a space
+#[derive(Encode, Decode, Clone, Eq, PartialEq, RuntimeDebug, TypeInfo, MaxEncodedLen)]
+pub struct SubscriptionSettings<Balance, RoleId> {
+    /// The balance required to subscribe to a space.
+    pub subscription: Balance,
+
+    /// Determines if subscriptions for a space s disabled.
+    pub disabled: bool,
+
+    /// The id of the role that will be granted for space subscriber.
+    pub role_id: RoleId,
```

Comment:
Why do we not put a permissions array here and create a role in extrinsic?

Reply:
Not here. Why should we store permissions set here if we'll store it in a role?
I think you meant to provide permissions set to an extrinsic, create a role using `RolesInterface`, and then put the role id here.